### ISSUE TITLE
Contribuindo no Github, dúvidas

### ISSUE THREAD

**Issue by RicardoRFaria** [Treeofsavior/PortugueseTranslation#29]
Pessoal, estou abrindo essa Issue apenas para discussão, para que possamos tirar dúvidas de vocês.
Vou iniciar com algumas dicas para que eu tenho visto que é motivo de dúvidas de muita gente.

O primeiro ponto importante é que o github não é uma plataforma ideal para traduções, ela é pra edição colaborativa de código, apesar de servir, não é a ideal por assim digamos.

Quem já contribuiu percebeu que  é necessário efetuar uma cópia do repositório atual do TOS, efetuar a modificação no seu repositório e depois solicitar a junção do conteúdo que é o que chamamos de pull request (PR).

Um PR é uma solicitação de junção do estado atual da sua "branch" que no caso da maioria tem sido a branch "master". 
Ao alterar qualquer coisa na branch que você está com um PR aberto, o conteúdo da alteração é automaticamente incluído no grupo de alterações do PR. Ou seja, se alguém lhe sugerir uma correção, basta aplicar e efetuar o commit, ela já vai diretamente para o PR.
O comportamente que descrevi acima é muito desejável quando as novas alterações fazem parte realmente daquele PR, ou seja, modificação do mesmo arquivo, mas caso seja de outro arquivo, ele é totalmente indesejado, conforme a IMCGames, cada PR deve conter a alteração de apenas um único arquivo.

Qual o caminho ideal? Ao efetuar o fork do projeto de tradução do TOS, você deve criar uma "branch", você pode verificar como criá-lo pesquisando na internet, efetua as alterações que você deseja abrir no PR apenas naquela branch. Posteriormente abra o PR através desta branch contendo as alterações.

E se eu for iniciar a tradução de outra parte? Crie outra branch com base no master e repita o processo acima.

Quaisquer dúvidas citem abaixo, tentarei ajudar assim que possível.

**Comment by RicardoRFaria** [comment]
Se os commits forem teus, vai tudo no mesmo pull request, se forem commits alheios, ai só a IMC pode fazer merge de tudo.

**Comment by heytrevor** [comment]
Entendi! Valeu, galera. Bem, pra evitar a fadiga e a possivel zona que isso possa gerar, deletei/fechei minha pull antiga e refiz da maneira que -acredito- ser a melhor.

**Comment by wevtimoteo** [comment]
Acho que vale colocarmos na wiki essas duas questões:

1) Arquivo correto a ser modificado (o BP ao invés do ENG)
2) Convenções a respeito dos nomes reservados que não devem ser traduzidos (algumas classes, Savior, etc)

O que acham? /cc @RicardoRFaria @heytrevor @BrasilToS

**Comment by heytrevor** [comment]
Concordo. Tá muito espalhado ainda as convenções. Até surgiu uma nova duvida aqui agora. Sei que Tree of Savior não será traduzido. Mas tenho traduzido "Savior" quando alguem lhe chama disso no jogo por Redentor. (achei válido/necessário). Mas um "Olá Savior!" também não cai mal..

**Comment by fabiohenm** [comment]
Pois é, isso está confundindo um bocado quem não está acostumado com o github como eu. Obrigado pela explicação.

**Comment by CrisInuyasha** [comment]
Estou tentando seguir o que foi recomendado aqui, mas agora no meu pull request apareceu que o branch contém conflitos que devem ser resolvidos (Estou com uma branch entre as linhas 1 e 200 e outra de 201 a 400 no arquivo ETC_BP.tsv). Isso é normal?

**Comment by heytrevor** [comment]
A IMC escreveu algo? Btw nas Guidelines eles recomendam que cada Pull Request tem de ter no MAXIMO 150 linhas de alteração. se as suas tem 200 linhas eles podem vetar...

**Comment by CrisInuyasha** [comment]
Verdade, nas próximas vou me atentar e enviar somente 150, obrigado pelo toque.

**Comment by heytrevor** [comment]
Não posso afirmar que seja isso, mas pode ser. De qualquer forma tu pode tentar fechar sua PR, fazer as correções (caso tenham) e aproveitar o que já ta pronto abrindo 2 ou 3 PR separadas e ver se fica ok. @CrisInuyasha

**Comment by sunlightning** [comment]
Eu criei um novo repositório meu(uma cópia do original) e editei duas linhas.
Aí fiz um pull request e fica como OPEN.

Só que aparece uma opção de MERGE. O que isso significa? Eu atualizo só o meu repositório ou modifico já o original deles? Não entendi direito.

**Comment by RicardoRFaria** [comment]
Uma vez aberto o Pull Request, a comunidade vai fazer críticas caso haja algo errado e posteriormente a equipe da IMCGames vai se encarregar de efetuar aceitar ou rejeitar o Pull Request.
Caso as linhas que você modificou já tenham sido modificadas, o próprio github lhe avisará que você deve fazer o merge.

**Comment by sunlightning** [comment]
Mas o merge serve para quê? Para eu atualizar meu próprio repositório?

**Comment by RicardoRFaria** [comment]
O merge serve nas duas vias, tanto para a IMCGames agrupar suas mudanças ao repositorio deles, quando para você atualizar seu repositório com as alterações presentes no repositório da IMCGames.
Aqui um link de como atualizar o seu fork com as atualizações da IMCGames via linha de comando.
https://help.github.com/articles/syncing-a-fork/
Com ferramentas como o tortoise git é mais simples.

**Comment by sunlightning** [comment]
O que eu quero saber é se eu posso dar MERGE logo depois da dar Pull Request para atualizar meu repositório ou só posso dar MERGE quando o Pull Request for aprovado.

**Comment by sunlightning** [comment]
Porque quando faço Pull Request, por exemplo, as modificações que fiz não ficam salvas no meu repositório, como se eu não tivesse modificado nada.

**Comment by RicardoRFaria** [comment]
Você está fazendo confusão cara, quando você faz pull request, as mudanças continuam no seu repositório normalmente, e nada lhe impede de sincronizar o seu fork com a da imc games durante este processo.

**Comment by sunlightning** [comment]
Merge então é para sincronizar com o repositório da IMC? É que eu não consegui entender muito bem o que o Merge faz.

**Comment by RicardoRFaria** [comment]
Merge junta dois projetos cara, ai depende do quem está fazendo. No caso da IMC, ela vai juntar o seu com o dela, e no seu caso, você vai juntar o dela com o seu.

**Comment by sunlightning** [comment]
Se o meu repositório está atualizado em relação ao deles, então não há necessidade de merge, certo?

**Comment by RicardoRFaria** [comment]
Isso.

**Comment by sunlightning** [comment]
Valeu.
